pkg/ccl/streamingccl/streamproducer/producer_job.go

```diff
@@ -105,13 +131,73 @@ func doStartReplicationStream(
 		return streaming.InvalidStreamID, errors.New("admin role required to start stream replication jobs")
 	}
 
-	registry := evalCtx.ExecConfigAccessor.JobRegistry().(*jobs.Registry)
+	registry := execConfig.JobRegistry
 	timeout := streamingccl.StreamReplicationJobLivenessTimeout.Get(&evalCtx.Settings.SV)
-	jobID, jr := makeProducerJobRecord(registry, tenantID, timeout, evalCtx.SessionData().User())
-	if _, err := registry.CreateAdoptableJobWithTxn(evalCtx.Ctx(), jr, jobID, txn); err != nil {
+	ptsID := uuid.MakeV4()
+	jr := makeProducerJobRecord(registry, tenantID, timeout, evalCtx.SessionData().User(), ptsID)
+	if _, err := registry.CreateAdoptableJobWithTxn(evalCtx.Ctx(), jr, jr.JobID, txn); err != nil {
 		return streaming.InvalidStreamID, err
 	}
-	return streaming.StreamID(jobID), nil
+
+	ptp := execConfig.ProtectedTimestampProvider
+	statementTime := hlc.Timestamp{
+		WallTime: evalCtx.GetStmtTimestamp().UnixNano(),
+	}
+	pts := jobsprotectedts.MakeRecord(ptsID, int64(jr.JobID), statementTime,
+		[]roachpb.Span{*makeTenantSpan(tenantID)}, jobsprotectedts.Jobs)
+
+	if err := ptp.Protect(evalCtx.Ctx(), txn, pts); err != nil {
+		return streaming.InvalidStreamID, err
+	}
+	return streaming.StreamID(jr.JobID), nil
+}
+
+// updateReplicationStreamProgress updates the job progress for an active replication
+// stream specified by 'streamID' and returns error if the stream is no longer active.
+func updateReplicationStreamProgress(
+	ctx context.Context,
+	expiration time.Time,
+	ptsProvider protectedts.Provider,
+	registry *jobs.Registry,
+	txn *kv.Txn,
+	streamID streaming.StreamID,
+	ts hlc.Timestamp,
+) error {
+	jr, err := registry.LoadJobWithTxn(ctx, jobspb.JobID(streamID), txn)
+	if err != nil {
+		return err
+	}
+
+	status, err := jr.CurrentStatus(ctx, txn)
+	if err != nil {
+		return err
+	}
+	if status != jobs.StatusRunning {
+		return errors.Errorf("job %d not running", streamID)
+	}
+
+	d := jr.Details().(jobspb.StreamReplicationDetails)
+	ptsRecord, err := ptsProvider.GetRecord(ctx, txn, *d.ProtectedTimestampRecord)
+	if err != nil {
+		return err
+	}
+
+	if shouldUpdatePTS := ptsRecord.Timestamp.
+		Add(streamingccl.MinProtectedTimestampUpdateAdvance.Nanoseconds(), 0).
+		Less(ts); shouldUpdatePTS {
```

Comment:
Maybe this MinProtectedTS advance is .. a bit premature optimization.
Perhaps we can simply update pts record to ts provided ts > ptsRecord.Timestamp?
We can add min advance later.